src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1657,6 +1658,13 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
             ;
             $container->setDefinition($transportId = 'messenger.transport.'.$name, $transportDefinition);
             $senderAliases[$name] = $transportId;
+
+            $transportRetryReferences[$name] = new Reference($transport['retry_strategy']['service']);
+            if ('messenger.retry.exponential_retry_strategy' === $transport['retry_strategy']['service']) {
+                $container->getDefinition('messenger.retry.exponential_retry_strategy')
```

Comment:
Im not sure I like this. This makes `exponential_retry_strategy` very special... 

- Should we add a list with all our supported strategies here? 
- How do I do if I want some config to my custom strategy?

Or maybe Im overthinking... This is how we configure our symfony services and yes, they should be treated special.

Reply:
We should validate the configuration so we cant use `max_retries` when we dont use `exponential_retry_strategy`

Reply:
> We should validate the configuration so we cant use max_retries when we dont use exponential_retry_strategy

Yes! This is already done

> Im not sure I like this. This makes exponential_retry_strategy very special...

I'm also not sure about this. By giving *one* strategy, it made configuration easy. If we also add a second strategy, for example, then we would also need to make its options configurable. So, I took the middle ground of: let's give them one nice strategy out-of-the-box.

> How do I do if I want some config to my custom strategy

Create a class that implements the interface, then set its id on the `retry_strategy.service` under the transport. So, retry strategies are transport-by-transport.

I'd also like to make the retry stuff configurable on a message-by-message basis. Perhaps there could be a stamp that the user adds that defines the retry behavior for a specific message.

Reply:
I've replaced the exponential retry strategy with a more generic, flexible one that should cover 95% of use-cases. The last 5% can register their own service for this.